tier1/convar.cpp

```diff
@@ -26,154 +26,214 @@
-void check_size() {
-	static_assert(ExpectedSize == RealSize, "Size mismatch");
-};
+void AddCommand( ConCommandCreation_t& cmd )
+{
+	if (ConCommandRegList::s_bConCommandsRegistered && s_bRegistered)
+	{
+		RegisterCommand( cmd );
+		return;
+	}
 
-class ConVarRegList;
-static ConVarRegList* s_pConVarRegList = nullptr;
+	if ( !s_pCommandRegList || s_pCommandRegList->Count() == 100 )
+	{
+		ConCommandRegList* newList = new ConCommandRegList;
+		newList->SetNextList( s_pCommandRegList );
 
-class ConVarRegList
+		s_pCommandRegList = newList;
+	}
+
+	s_pCommandRegList->Add( cmd );
+}
+
+void RegisterConVar( ConVarCreation_t& cvar )
 {
-public:
-	ConVarRegList()
+	g_pCVar->RegisterConVar( cvar, s_nCVarFlag, cvar.refHandle, cvar.refConVar );
+	if (!cvar.refHandle->IsValid())
 	{
-		check_size<ConVar, 0x60>();
-		check_size<ConVarRegList, 11216>();
+		Plat_FatalErrorFunc( "RegisterConVar: Unknown error registering convar \"%s\"!\n", cvar.name );
+		DebuggerBreakIfDebugging();
 	}
+}
 
-	static bool AreConVarsRegistered()
+void UnRegisterConVar( ConVarHandle& cvar )
+{
+	if (cvar.IsValid())
 	{
-		return s_bConVarsRegistered;
+		g_pCVar->UnregisterConVar( cvar );
+
+		cvar.Invalidate();
 	}
+}
+
+class ConVarRegList;
+static ConVarRegList* s_pConVarRegList = nullptr;
+
+class ConVarRegList
+{
+public:
+	ConVarRegList() {}
 
 	static void RegisterAll()
 	{
-		if (!s_bConVarsRegistered && g_pCVar)
+		if ( !s_bConVarsRegistered && g_pCVar )
 		{
 			s_bConVarsRegistered = true;
 
-			ConVarRegList* pList = s_pConVarRegList;
-			while (pList != nullptr)
+			ConVarRegList* list = s_pConVarRegList;
+			while ( list )
 			{
-				FOR_EACH_VEC(s_pConVarRegList->m_Vec, i)
+				FOR_EACH_VEC( list->m_Vec, i )
 				{
-					ConVar* pConVar = &pList->m_Vec[i];
-					ConVarHandle hndl;
-					//g_pCVar->RegisterConVar(pConVar, s_nCVarFlag, hndl);
-					//pConVar->SetHandle(hndl);
-
-					if (!hndl.IsValid())
-					{
-						Plat_FatalErrorFunc("RegisterConCommand: Unknown error registering convar \"%s\"!\n", pConVar->GetName());
-						DebuggerBreakIfDebugging();
-					}
+					RegisterConVar( list->m_Vec[i] );
 				}
 
-				ConVarRegList *pNext = pList->m_pNext;
-				delete pList;
-				pList = pNext;
+				ConVarRegList *pNext = list->m_pNext;
+				delete list;
+				list = pNext;
 			}
-
-			s_pConVarRegList = nullptr;
 		}
 	}
+
 private:
-	CUtlVectorFixed<ConVar, 100> m_Vec;
-	ConVarRegList* m_pNext = nullptr;
+	friend void SetupConVar( ConVarCreation_t& cvar );
+
+	void SetNextList( ConVarRegList* list )
+	{
+		m_pNext = list;
+	}
 
+	int Count() const
+	{
+		return m_Vec.Count();
+	}
+
+	void Add( const ConVarCreation_t& cvar )
+	{
+		m_Vec.AddToTail( cvar );
+	}
+
+	CUtlVectorFixed<ConVarCreation_t, 100> m_Vec;
+	ConVarRegList* m_pNext = nullptr;
+public:
 	static bool s_bConVarsRegistered;
 };
+static_assert(sizeof(ConVarRegList) == 0x2BD0, "Size mismatch");
 
 bool ConVarRegList::s_bConVarsRegistered = false;
-#endif // CONVAR_WORK_FINISHED
+
+// sub_10B79F0

```

Comment:
These comments pointing to some ida funcs and refs aren't really much of a use since they'd get obsoleted quite quickly with newer binaries.